### PR TITLE
E2E Redirect Shopper

### PR DESCRIPTION
**Pull request by maassenbas**
E2E tests for Redirect Shopper
- Klarna
- iDeal
- EPS
- Giropay
- Oney
- Trustly
- Vipps
- MobilePay
## Tested scenarios

**Fixed issue**:

### CODE REVIEW

File: tests/e2e/fixtures/EUR.js [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+import CheckoutPage from "../pages/CheckoutPage";
+import testNetherlands from "./countriesEUR/NL";
+import testFrance from "./countriesEUR/FR";
+import testGermany from "./countriesEUR/DE";
+import testAustria from "./countriesEUR/AT";
 import { regionsEnum } from "../data/enums"
-//TODO check if we want to move URL
+
 fixture`EUR`
     .page(`https://${process.env.SFCC_HOSTNAME}/s/RefArch/home?lang=${regionsEnum.EU}`)
```
Comment (wboereboom) [inline tests/e2e/fixtures/EUR.js:9]: region does not need to be in this URL as we change locales during checkout

Comment (wboereboom) [inline tests/e2e/fixtures/EUR.js:9]:
```suggestion
    .page(`https://${process.env.SFCC_HOSTNAME}/s/RefArch/home`)
```

File: tests/e2e/fixtures/NOK.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    await doTrustlyPayment(true);
+    await checkoutPage.expectSuccess();
+  });
+
+  test('Trustly Fail', async () => {
+    await doTrustlyPayment(false);
+    await checkoutPage.expectRefusal();
+  });
+
+
```
Comment (wboereboom) [inline tests/e2e/fixtures/NOK.js:42]:
```suggestion
```

File: tests/e2e/fixtures/NOK.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
```
Comment (wboereboom) [inline tests/e2e/fixtures/NOK.js:43]:
```suggestion
```

File: tests/e2e/fixtures/SEK.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      await checkoutPage.goToCheckoutPageWithFullCart(regionsEnum.DK);
+      await checkoutPage.setShopperDetails(shopperData.DK);
+    });
+
+test('Swish', async () => {
+
+});
+
+
+
```
Comment (wboereboom) [inline tests/e2e/fixtures/SEK.js:26]:
```suggestion
```

File: tests/e2e/fixtures/SEK.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      checkoutPage = new CheckoutPage();
+      await checkoutPage.goToCheckoutPageWithFullCart(regionsEnum.DK);
+      await checkoutPage.setShopperDetails(shopperData.DK);
+    });
+
+test('Swish', async () => {
+
+});
+
+
```
Comment (wboereboom) [inline tests/e2e/fixtures/SEK.js:25]:
```suggestion
```

File: tests/e2e/fixtures/USD.js [hunk 6]
```diff
@@ -1,10 +1,20 @@
 import { regionsEnum } from "../data/enums";
+import { doAffirmPayment } from "../paymentFlows/redirectShopper";
+import CheckoutPage from "../pages/CheckoutPage";
+
+const shopperData = require("../data/shopperData.json");
 
 fixture`USD`
     .page(`https://${process.env.SFCC_HOSTNAME}/s/RefArch/home?lang=${regionsEnum.US}`)
```
Comment (wboereboom) [inline tests/e2e/fixtures/USD.js:8]:
```suggestion
    .page(`https://${process.env.SFCC_HOSTNAME}/s/RefArch/home`)
```

File: tests/e2e/pages/PaymentMethodsPage.js [hunk 7]
```diff
@@ -18,4 +20,145 @@ export default class PaymentMethodsPage {
...
+
+        await t
+            .click(Selector('#rb_facilypay_4x'))
+            .click(oneyGender)
+            .click(oneyDateOfBirth)
+            .typeText(oneyDateOfBirth, shopper.dateOfBirth)
+            .typeText(oneyEmail, shopper.shopperEmail)
+    }
+
+    confirmOneyPayment = async () => {
```
Comment (wboereboom) [inline tests/e2e/pages/PaymentMethodsPage.js:37]: can this go?

File: tests/e2e/pages/PaymentMethodsPage.js [hunk 8]
```diff
@@ -18,4 +20,145 @@ export default class PaymentMethodsPage {
...
+    }
+
+    initiateAffirmPayment = async (shopper) => {
+        const affirmEmail = Selector('#component_affirm input[name="shopperEmail"]');
+        await t
+            .click(Selector('#rb_affirm'))
+            .typeText(affirmEmail, shopper.shopperEmail)
+    }
+
+    confirmSimulator = async () => {
```
Comment (wboereboom) [inline tests/e2e/pages/PaymentMethodsPage.js:114]: what are these methods for?

**Review by wboereboom — CHANGES_REQUESTED** [review]

File: tests/e2e/fixtures/countriesEUR/DE.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        await doKlarnaAccountPayment(true);
+        await checkoutPage.expectSuccess();
+    });
+
+    test('Klarna Account Fail', async t => {
+        await checkoutPage.setShopperDetails(shopperData.DE);
+        await doKlarnaAccountPayment(false);
+        await checkoutPage.expectRefusal();
+    });
+    test.only('Giropay Success', async t => {
```
Comment (wboereboom) [inline tests/e2e/fixtures/countriesEUR/DE.js:42]: the pipeline now **only** tests Giropay Success

```suggestion
    test('Giropay Success', async t => {
```

File: tests/e2e/pages/PaymentMethodsPage.js (hunk 8 above)
Comment (maassenbas) [reply tests/e2e/pages/PaymentMethodsPage.js:114]: These methods are for the simulator from SimulatorAcquirerAccount, which is used for multiple payment methods (e.g. EPS). Although, currently this simulator is down, so can't complete the payment. Will have to update it once the simulator works again. Will reach out internally regarding this.

File: tests/e2e/fixtures/EUR.js (hunk 1 above)
Comment (maassenbas) [reply tests/e2e/fixtures/EUR.js]: Yes, makes sense. Will apply it.

**Review by zaiddreakh — APPROVED** [review]

**Pull request merged**